### PR TITLE
[h264][h265]implemented ROI_Priority => ROI_DeltaQP conversion

### PR DESCRIPTION
**Pull request by AntonGrishin**
implemented the conversion ROI_Priority => ROI_Delta QP without
external changes in library behavior.

Change-Id: If4c658142769329f00a19e3554a94bfeb956062d
I9a46dd302e2d3a40a61a51d4150de6fd9112ab8a
I3dcaf9ae439676bb165429d2450ab67d40637708

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #include "mfx_h265_encode_hw_ddi.h"
 #include <assert.h>
 #include <math.h>
+#include <iostream>
```
Comment (eistarov) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: what for?

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 2]
```diff
@@ -2251,25 +2251,21 @@ mfxStatus MfxHwH264Encode::CheckAndFixRoiQueryLike(
     mfxRoiDesc *          roi,
     mfxU16                roiMode)
 {
-    mfxStatus checkSts = CheckAndFixOpenRectQueryLike(par, (mfxRectDesc*)roi);
+   mfxStatus checkSts = CheckAndFixOpenRectQueryLike(par, (mfxRectDesc*)roi);
```
Comment (eistarov) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: wrong indent

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 3]
```diff
@@ -4321,34 +4317,21 @@ mfxStatus MfxHwH264Encode::CheckVideoParamQueryLike(
...
     {
         unsupported = true;
         extRoi->NumROI = 0;
     }
 
-    if (extRoi->NumROI && par.mfx.RateControlMethod != MFX_RATECONTROL_CQP &&
-        extRoi->ROIMode == MFX_ROI_MODE_QP_DELTA && hwCaps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)
+    if ((extRoi->NumROI && (extRoi->ROIMode == MFX_ROI_MODE_QP_DELTA ||
+        extRoi->ROIMode == MFX_ROI_MODE_PRIORITY ))&&
+        hwCaps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)
```
Comment (eistarov) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: Senseless over-complication, ROIMode value already verified above.
This check is enough now:
if (extRoi->NumROI && hwCaps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 4]
```diff
@@ -6576,27 +6555,18 @@ mfxStatus MfxHwH264Encode::CheckRunTimeExtBuffers(
...
-                extRoi->ROIMode == MFX_ROI_MODE_QP_DELTA && caps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)
+            if (extRoi->ROIMode != MFX_ROI_MODE_QP_DELTA && extRoi->ROIMode != MFX_ROI_MODE_PRIORITY)
             {
                 checkSts = MFX_WRN_INCOMPATIBLE_VIDEO_PARAM;
                 actualNumRoi = 0;
             }
 
-            if (video.mfx.RateControlMethod != MFX_RATECONTROL_CQP &&
-                extRoi->ROIMode == MFX_ROI_MODE_PRIORITY && caps.ddi_caps.ROIBRCPriorityLevelSupport == 0)
+            if (extRoi->ROIMode == MFX_ROI_MODE_QP_DELTA && caps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0)
```
Comment (eistarov) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: video.mfx.RateControlMethod != MFX_RATECONTROL_CQP && caps.ddi_caps.ROIBRCDeltaQPLevelSupport == 0

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp (hunk 1 above)
Comment (AntonGrishin) [reply _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: Sorry, forgot remove after debug

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 2 above)
Comment (AntonGrishin) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: done

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 3 above)
Comment (AntonGrishin) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: done

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 4 above)
Comment (AntonGrishin) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: done

**Pull request merged**